docs/design/pattern_matching.md

```diff
@@ -126,9 +128,28 @@ A name binding pattern is a pattern.
 -   _binding-pattern_ ::= _identifier_ `:` _expression_
 -   _proper-pattern_ ::= _binding-pattern_
 
-The _identifier_ specifies the name of the _binding_. The type of the binding is
-specified by the _expression_. The scrutinee is implicitly converted to that
-type if necessary. The binding is then _bound_ to the converted value.
+A name binding pattern declares a _binding_ with a name specified by the
+_identifier_, which can be used as an expression. If the binding pattern is
+enclosed by a `var` pattern, it is a _reference binding pattern_, and the
+binding is a durable reference expression. Otherwise, it is a _value binding
+pattern_, and the binding is a value expression.
+
+A _variable binding pattern_ is a special kind of reference binding pattern,
+where the enclosing `var` pattern doesn't have a tuple or struct subpattern.
```

Comment:
Is this equivalent to saying that a variable binding pattern is a binding pattern that is directly contained within a `var` pattern, or is there some other kind of intermediate pattern that would be allowed here while still forming a variable binding pattern? (Perhaps an alternative pattern or something like an `addr` pattern?)